native: implicit declaration warnings

Compiling the `default` example application for native throws several warnings due to missing includes:

```
board_config.c: In function ‘config_load’:
board_config.c:39:5: warning: implicit declaration of function ‘getpid’ [-Wimplicit-function-declaration]
     int pid = getpid();
     ^
make[2]: Entering directory '/home/oleg/git/RIOT/boards/native/drivers'
native-uart0.c: In function ‘uart0_puts’:
native-uart0.c:60:17: warning: implicit declaration of function ‘write’ [-Wimplicit-function-declaration]
                 (nwritten = write(
                 ^
native-uart0.c: In function ‘init_unix_socket’:
native-uart0.c:151:5: warning: implicit declaration of function ‘getpid’ [-Wimplicit-function-declaration]
     snprintf(sa.sun_path, sizeof(sa.sun_path), "/tmp/riot.tty.%d", getpid());
     ^
native-uart0.c:152:5: warning: implicit declaration of function ‘unlink’ [-Wimplicit-function-declaration]
     unlink(sa.sun_path); /* remove stale socket */
     ^
native-uart0.c: In function ‘handle_uart_in’:
native-uart0.c:171:5: warning: implicit declaration of function ‘read’ [-Wimplicit-function-declaration]
     nread = read(STDIN_FILENO, buf, sizeof(buf));
     ^
native-uart0.c:179:17: warning: implicit declaration of function ‘dup2’ [-Wimplicit-function-declaration]
                 if (dup2(_native_null_out_file, STDOUT_FILENO) == -1) {
                 ^
make[2]: Leaving directory '/home/oleg/git/RIOT/boards/native/drivers'
make[1]: Leaving directory '/home/oleg/git/RIOT/boards/native'
"make" -C /home/oleg/git/RIOT
make[1]: Entering directory '/home/oleg/git/RIOT'
mkdir -p /home/oleg/git/RIOT/examples/default/bin/native/
make[2]: Entering directory '/home/oleg/git/RIOT/cpu/native'
irq_cpu.c: In function ‘native_interrupt_init’:
irq_cpu.c:507:5: warning: implicit declaration of function ‘pipe’ [-Wimplicit-function-declaration]
     if (pipe(_sig_pipefd) == -1) {
     ^
native_cpu.c: In function ‘reboot_arch’:
native_cpu.c:72:5: warning: implicit declaration of function ‘execve’ [-Wimplicit-function-declaration]
     if (execve(_native_argv[0], _native_argv, NULL) == -1) {
     ^
startup.c: In function ‘_native_null_in’:
startup.c:54:5: warning: implicit declaration of function ‘pipe’ [-Wimplicit-function-declaration]
     if (pipe(_native_null_in_pipe) == -1) {
     ^
startup.c:62:5: warning: implicit declaration of function ‘dup2’ [-Wimplicit-function-declaration]
     if (dup2(_native_null_in_pipe[0], STDIN_FILENO) == -1) {
     ^
startup.c: In function ‘_native_log_stdout’:
startup.c:88:9: warning: implicit declaration of function ‘getpid’ [-Wimplicit-function-declaration]
         snprintf(stdout_logname, sizeof(stdout_logname), "/tmp/riot.stdout.%d", getpid());
         ^
startup.c: In function ‘daemonize’:
startup.c:141:5: warning: implicit declaration of function ‘fork’ [-Wimplicit-function-declaration]
     if ((pid = fork()) == -1) {
     ^
make[3]: Entering directory '/home/oleg/git/RIOT/cpu/native/rtc'
make[3]: Leaving directory '/home/oleg/git/RIOT/cpu/native/rtc'
make[3]: Entering directory '/home/oleg/git/RIOT/cpu/native/net'
tap.c: In function ‘send_buf’:
tap.c:222:5: warning: implicit declaration of function ‘write’ [-Wimplicit-function-declaration]
     if ((nsent = write(_native_tap_fd, buf, to_send)) == -1) {;
     ^
tap.c: In function ‘tap_init’:
tap.c:308:5: warning: implicit declaration of function ‘getpid’ [-Wimplicit-function-declaration]
     if (fcntl(_native_tap_fd, F_SETOWN, getpid()) == -1) {
```

--- a/boards/native/Makefile
+++ b/boards/native/Makefile
@@ -7,10 +7,11 @@ all: $(BINDIR)$(MODULE).a
 
 include $(RIOTBASE)/Makefile.base
 
-$(BINDIR)%.o: %.c
-	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$*.o
+$(BINDIR)$(MODULE)/%.o: %.c
+	@mkdir -p $(BINDIR)$(MODULE)
+	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$(MODULE)/$*.o
 	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -MM $*.c |\
-		sed -e "1s|^|$(BINDIR)|" > $(BINDIR)$*.d
+		sed -e "1s|^|$(BINDIR)$(MODULE)/|" > $(BINDIR)$(MODULE)/$*.d
 
 clean::
 	@for i in $(DIRS) ; do "$(MAKE)" -C $$i clean || exit 1; done ;

--- a/boards/native/drivers/Makefile
+++ b/boards/native/drivers/Makefile
@@ -5,7 +5,8 @@ all: $(BINDIR)$(MODULE).a
 
 include $(RIOTBASE)/Makefile.base
 
-$(BINDIR)%.o: %.c
-	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$*.o
+$(BINDIR)$(MODULE)/%.o: %.c
+	@mkdir -p $(BINDIR)$(MODULE)
+	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$(MODULE)/$*.o
 	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -MM $*.c |\
-		sed -e "1s|^|$(BINDIR)|" > $(BINDIR)$*.d
+		sed -e "1s|^|$(BINDIR)$(MODULE)/|" > $(BINDIR)$(MODULE)/$*.d

--- a/cpu/native/Makefile
+++ b/cpu/native/Makefile
@@ -13,10 +13,11 @@ all: $(BINDIR)$(MODULE).a
 
 include $(RIOTBASE)/Makefile.base
 
-$(BINDIR)%.o: %.c
-	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$*.o
+$(BINDIR)$(MODULE)/%.o: %.c
+	@mkdir -p $(BINDIR)$(MODULE)
+	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$(MODULE)/$*.o
 	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -MM $*.c |\
-		sed -e "1s|^|$(BINDIR)|" > $(BINDIR)$*.d
+		sed -e "1s|^|$(BINDIR)$(MODULE)/|" > $(BINDIR)$(MODULE)/$*.d
 
 clean::
 	@for i in $(DIRS) ; do "$(MAKE)" -C $$i clean || exit 1; done ;

--- a/cpu/native/net/Makefile
+++ b/cpu/native/net/Makefile
@@ -2,7 +2,8 @@ MODULE = nativenet
 
 include $(RIOTBASE)/Makefile.base
 
-$(BINDIR)%.o: %.c
-	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$*.o
+$(BINDIR)$(MODULE)/%.o: %.c
+	@mkdir -p $(BINDIR)$(MODULE)
+	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$(MODULE)/$*.o
 	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -MM $*.c |\
-		sed -e "1s|^|$(BINDIR)|" > $(BINDIR)$*.d
+		sed -e "1s|^|$(BINDIR)$(MODULE)/|" > $(BINDIR)$(MODULE)/$*.d

--- a/cpu/native/rtc/Makefile
+++ b/cpu/native/rtc/Makefile
@@ -2,7 +2,8 @@ MODULE =rtc
 
 include $(RIOTBASE)/Makefile.base
 
-$(BINDIR)%.o: %.c
-	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$*.o
+$(BINDIR)$(MODULE)/%.o: %.c
+	@mkdir -p $(BINDIR)$(MODULE)
+	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -c $*.c -o $(BINDIR)$(MODULE)/$*.o
 	$(AD)$(CC) $(CFLAGS) $(NATIVEINCLUDES) -MM $*.c |\
-		sed -e "1s|^|$(BINDIR)|" > $(BINDIR)$*.d
+		sed -e "1s|^|$(BINDIR)$(MODULE)/|" > $(BINDIR)$(MODULE)/$*.d

